orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/model/TemplateConfiguration.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class TemplateConfiguration {
+
+  private String schema;
+  private String id;
+  private PipelineDefinition pipeline;
+  private PipelineConfiguration configuration;
+  private List<StageDefinition> stages;
+  private List<TemplateModule> modules;
+
+  private final String runtimeId = UUID.randomUUID().toString();
+
+  public static class PipelineDefinition {
+
+    private String application;
+    private String name;
+    private TemplateSource template;
+    private Map<String, Object> variables;
+
+    public String getApplication() {
+      return application;
+    }
+
+    public void setApplication(String application) {
+      this.application = application;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public void setName(String name) {
+      this.name = name;
+    }
+
+    public TemplateSource getTemplate() {
+      return template;
+    }
+
+    public void setTemplate(TemplateSource template) {
+      this.template = template;
+    }
+
+    public Map<String, Object> getVariables() {
+      return variables;
+    }
+
+    public void setVariables(Map<String, Object> variables) {
+      this.variables = variables;
+    }
+  }
+
+  public static class TemplateSource {
+
+    private String source;
+
+    public String getSource() {
+      return source;
+    }
+
+    public void setSource(String source) {
+      this.source = source;
+    }
+  }
+
+  public static class PipelineConfiguration {
+
+    private List<String> inherit;
+    private Map<String, Object> concurrentExecutions;
+    private List<NamedHashMap> triggers;
+    private List<NamedHashMap> parameters;
+    private List<NamedHashMap> notifications;
+    private String description;
+
+    public List<String> getInherit() {
+      return inherit;
+    }
+
+    public void setInherit(List<String> inherit) {
+      this.inherit = inherit;
+    }
+
+    public Map<String, Object> getConcurrentExecutions() {
```

Comment:
Minor comment here. IMHO, it would be beneficial from a user's standpoint, as well as to avoid errors, to have this come out as a ```[String:Bool]```.  So for example ```onStageFailure``` we would have ```haltEntirePiple: true```.
This allows the user to just enable the option by passing true.